### PR TITLE
[New Rule] Command shell activity started via rundll32

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/390

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/windows/execution_command_shell_via_rundll32.toml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+name = "Command shell activity started via rundll32"
+risk_score = 21
+rule_id = "9ccf3ce0-0057-440a-91f5-870c6ad39093"
+severity = "low"
+tags = ["Elastic", "Windows"]
+type = "eql"
+
+query = '''
+process where event.type in ("start", "process_started") and
+ wildcard(process.name, "cmd.exe", "powershell.exe") and
```
Comment (rw-access) [inline rules/windows/execution_command_shell_via_rundll32.toml:23]: I've tried something like this before but it was noisy.
Any ideas to make this higher fidelity? If not, that's fine, we can capture its noise level with metadata.

Comment (Samirbous) [reply rules/windows/execution_command_shell_via_rundll32.toml:23]: I've used this before as a hunt in prod, usually main FP is related to rundll32 with cmdline value like "*\\SHELL32.dll*RunAsNewUser_RunDLL*" (result of right-click runas new user of cmd), few other exception is when EDR does not capture parent commanding (e.g. NA or NULL). As you suggested we can play and tune it with other attributes once we have alert telemetr.

**Comment by bm11100** [comment]
Do you think this will overlap with the current rule of  `Unusual Child Processes of RunDLL32`?

File: rules/windows/execution_command_shell_via_rundll32.toml (hunk 1 above)
Comment (rw-access) [reply rules/windows/execution_command_shell_via_rundll32.toml:23]: i would expect a lot more results matching the rundll32 -> cmd chain

Comment (Samirbous) [reply rules/windows/execution_command_shell_via_rundll32.toml:23]: do you have example of mayches other than the RunAsNewUser_RunDLL one ?, used this before in prod with more than 3000 endpoints and that was the only main match (90% of cases). given the fact that CS and other backdoors default to rundll32 and their cmd|interactive shell command defaults to cmd.exe its worth alerting (ratio viz vs noise is very acceptable).

**Comment by Samirbous** [comment]
> Do you think this will overlap with the current rule of `Unusual Child Processes of RunDLL32`? Will there be minimal command line parameters?

the Unusual Child Process looks for rundll32 with "no" cmdline (process.args_count < 2) and its not triggering on tests I did,

File: rules/windows/execution_command_shell_via_rundll32.toml [hunk 2]
```diff
@@ -0,0 +1,54 @@
+[metadata]
+creation_date = "2020/10/19"
+ecs_version = ["1.6.0"]
+maturity = "production"
+updated_date = "2020/10/19"
+
+[rule]
+author = ["Elastic"]
+description = "Identifies command shell activity started via rundll32 which is usually abused by attackers to host malicious code."
```
Comment (threat-punter) [inline rules/windows/execution_command_shell_via_rundll32.toml:9]:
```suggestion
description = "Identifies command shell activity started via rundll32 which is commonly abused by attackers to host malicious code."
```

File: rules/windows/execution_command_shell_via_rundll32.toml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+from = "now-9m"
+index = ["winlogbeat-*", "logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License"
+name = "Command shell activity started via rundll32"
```
Comment (threat-punter) [inline rules/windows/execution_command_shell_via_rundll32.toml:14]:
```suggestion
name = "Command Shell Activity Started via RunDLL32"
```

File: rules/windows/execution_command_shell_via_rundll32.toml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+severity = "low"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Execution"]
+type = "eql"
+
+query = '''
+process where event.type in ("start", "process_started") and
+ wildcard(process.name, "cmd.exe", "powershell.exe") and
+  process.parent.name == "rundll32.exe" and 
+  /* common FPs can be added here */
+   process.parent.args != "C:\\Windows\\System32\\SHELL32.dll,RunAsNewUser_RunDLL"
```
Comment (threat-punter) [inline rules/windows/execution_command_shell_via_rundll32.toml:26]:
```suggestion
process where event.type in ("start", "process_started") and
 wildcard(process.name, "cmd.exe", "powershell.exe") and
  process.parent.name == "rundll32.exe" and 
  /* common FPs can be added here */
  process.parent.args != "C:\\Windows\\System32\\SHELL32.dll,RunAsNewUser_RunDLL"
```

Comment (threat-punter) [inline rules/windows/execution_command_shell_via_rundll32.toml:26]:
```suggestion
  process.parent.args != "C:\\Windows\\System32\\SHELL32.dll,RunAsNewUser_RunDLL"
```

**Review by threat-punter — APPROVED** [review]
👍

File: rules/windows/execution_command_shell_via_rundll32.toml [hunk 5]
```diff
@@ -0,0 +1,53 @@
+[metadata]
+creation_date = "2020/10/19"
+ecs_version = ["1.6.0"]
+maturity = "production"
+updated_date = "2020/10/19"
+
+[rule]
+author = ["Elastic"]
+description = "Identifies command shell activity started via RunDLL32 which is commonly abused by attackers to host malicious code."
```
Comment (bm11100) [inline rules/windows/execution_command_shell_via_rundll32.toml:9]:
```suggestion
description = "Identifies command shell activity started via RunDLL32, which is commonly abused by attackers to host malicious code."
```

**Review by bm11100 — APPROVED** [review]

File: rules/windows/execution_command_shell_via_rundll32.toml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+severity = "low"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Execution"]
+type = "eql"
+
+query = '''
+process where event.type in ("start", "process_started") and
+ wildcard(process.name, "cmd.exe", "powershell.exe") and
+  process.parent.name == "rundll32.exe" and 
+  /* common FPs can be added here */
+  process.parent.args != "C:\\Windows\\System32\\SHELL32.dll,RunAsNewUser_RunDLL"
```
Comment (brokensound77) [inline rules/windows/execution_command_shell_via_rundll32.toml:26]:
```suggestion
process where event.type in ("start", "process_started") and
 process.name : ("cmd.exe", "powershell.exe") and
  process.parent.name : "rundll32.exe" and 
  /* common FPs can be added here */
  not process.parent.args : "C:\\Windows\\System32\\SHELL32.dll,RunAsNewUser_RunDLL"
```

**Pull request merged**